graphstorm-processing/graphstorm_processing/config/config_conversion/gconstruct_converter.py

```diff
@@ -0,0 +1,189 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from .converter_base import ConfigConverter
+from .meta_configuration import NodeConfig, EdgeConfig
+
+
+class GConstructConfigConverter(ConfigConverter):
+    """The Config Converter from GConstruct to GSProcessing
+
+    Parameters
+    ----------
+    data: valid GConstruct Config including nodes and edge config
+    ----------
+    """
+
+    @staticmethod
+    def _convert_label(labels: list[dict]) -> list[dict]:
+        """Convert labels config
+
+        Parameters
+        ----------
+        labels: list[dict]
+            The label information in the GConstruct format
+
+        Returns
+        -------
+        list[dict]
+            The label information in the GSProcessing format
+        """
+        labels_list = []
+        if labels in [[], [{}]]:
+            return []
+        for label in labels:
+            try:
+                label_column = label["label_col"]
+                label_type = label["task_type"]
+                label_dict = {"column": label_column, "type": label_type}
+                if "split_pct" in label:
+                    label_splitrate = label["split_pct"]
+                    # check if split_pct is valid
+                    assert sum(label_splitrate) == 1.0, "sum of the label split rate should be 1.0"
```

Comment:
Do we want the sum of splits == 1.0 in gsprocessing? In gconstruct, the sum can < 1.0 or be 0.
If it is zero, gconstruct will not automatically create train/val/test mask: https://github.com/awslabs/graphstorm/blob/main/python/graphstorm/gconstruct/transform.py#L1145-L1150